go-controller/pkg/ovn/ovn.go

```diff
@@ -564,28 +629,36 @@ func (oc *Controller) WatchPods() {
 				}
 				return
 			}
+			needsRetry := false
 			if podScheduled(pod) {
 				if err := oc.addLogicalPort(pod); err != nil {
 					klog.Errorf(err.Error())
 					oc.recordPodEvent(err, pod)
-					retryPods.Store(pod.UID, true)
+					needsRetry = true
 				}
 			} else {
 				// Handle unscheduled pods later in UpdateFunc
-				retryPods.Store(pod.UID, true)
+				needsRetry = true
+			}
+			if needsRetry {
+				oc.retryPodsLock.Lock()
+				defer oc.retryPodsLock.Unlock()
+				oc.retryPods[pod.UID] = retryEntry{pod, time.Now()}
 			}
 		},
 		UpdateFunc: func(old, newer interface{}) {
 			oldPod := old.(*kapi.Pod)
 			pod := newer.(*kapi.Pod)
 
-			_, retry := retryPods.Load(pod.UID)
-			if podScheduled(pod) && retry && util.PodWantsNetwork(pod) {
+			// if pod is in retry map, delete it we will handle the latest version here
```

Comment:
I don't know what this comment means

Reply:
it means with oc.checkAndDeleteRetryPod, we will delete the entry out of the retry map, so that we dont need to hold the entire lock of the map while we update the pod. This also guarantees that the goroutine retryLoop wont try to addLogicalPort for this pod at the same time while we are operating on it here in the Update handler of the controller

Reply:
I'll change the comment to make this more clear